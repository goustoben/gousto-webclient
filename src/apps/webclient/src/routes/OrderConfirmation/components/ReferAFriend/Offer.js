import React from 'react'
import PropTypes from 'prop-types'
import { config } from './config'
import css from './ReferAFriend.css'

const propTypes = {
  isYourOffer: PropTypes.bool.isRequired,
  offer: PropTypes.string.isRequired,
  theirMonthOffer: PropTypes.string,
}

const defaultProps = {
  theirMonthOffer: '',
}

const Offer = ({ isYourOffer, offer, theirMonthOffer }) => {
  const offerConfig = config(theirMonthOffer)
  const title = isYourOffer ? offerConfig.youGetTitle : offerConfig.yourFriendGetsTitle
  const description = isYourOffer ? offerConfig.youGetDescription : offerConfig.yourFriendGetsDescription

  return (
    <section className={css.offerWrapper}>
      <div className={css.offerTitle}>{title}</div>
      <div className={css.offer}>{offer}</div>
      <div className={css.offerDescription}>{description}</div>
    </section>
  )
}

Offer.propTypes = propTypes

Offer.defaultProps = defaultProps

export { Offer }
