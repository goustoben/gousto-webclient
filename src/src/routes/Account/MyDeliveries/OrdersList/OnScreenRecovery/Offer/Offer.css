@value Haricot, Carrot, White from 'styles/colors.css';
@value axiformaFontStack from 'styles/typography.css';

.offerWrapper {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.discountOSR {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  height: 78.44px;
  width: 78.77px;
  border-radius: 50%;
  box-sizing: border-box;
  line-height: 21px;
}

.discountOSROrder {
  border: 1.44px solid Haricot;
  background-color: Haricot;
}

.discountOSRSubscription {
  border: 1.44px solid Carrot;
  background-color: Carrot;
}

.discountOSR__container {
  display: flex;
  flex-direction: column;
  max-width: 60px;
  font-size: 22px;
  font-family: axiformaFontStack;
  font-weight: bold;
  color: White;
}

.discountOSR__sub {
  font-size: 70%;
  line-height: 1.2;
}

.messageDiscountOSR {
  font-family: Avenir;
  font-size: 16px;
  line-height: 21px;
  text-align: left;
  margin-left: 20px;
}

.messageDiscountOSROrder {
  color: Haricot;
}

.messageDiscountOSRSubscription {
  color: Carrot;
}
