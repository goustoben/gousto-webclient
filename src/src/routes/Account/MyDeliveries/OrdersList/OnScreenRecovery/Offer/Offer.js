import PropTypes from 'prop-types'
import classnames from 'classnames'
import React from 'react'

import css from './Offer.css'

import { ORDER_TYPE, SUBSCRIPTION_TYPE } from '../config'

const propTypesOffer = {
  offer: PropTypes.shape({
    formattedValue: PropTypes.string,
    rawMessage: PropTypes.shape({
      text: PropTypes.string,
      values: PropTypes.arrayOf(PropTypes.shape({
        key: PropTypes.string,
        value: PropTypes.string,
      }))
    })
  }),
  type: PropTypes.oneOf([SUBSCRIPTION_TYPE, ORDER_TYPE])
}

const defaultProps = {
  offer: {}
}

const transformMessage = (text, values) => {
  let formattedText = text

  values.forEach(item => {
    formattedText = formattedText.replace(`{:${item.key}:}`, `<strong>${item.value}</strong>`)
  })

  return formattedText
}

const getDiscountOSRClass = (type) => {
  switch (type) {
  case ORDER_TYPE:
    return 'discountOSROffer'
  case SUBSCRIPTION_TYPE:
    return 'discountOSRSubscription'
  default:
    return 'discountOSROffer'
  }
}

const getMessageDiscountOSRClass = (type) => {
  switch (type) {
  case ORDER_TYPE:
    return 'messageDiscountOSROffer'
  case SUBSCRIPTION_TYPE:
    return 'messageDiscountOSRSubscription'
  default:
    return 'messageDiscountOSROffer'
  }
}

const Offer = ({ offer, type }) => {
  if (!(offer && offer.message)) return null

  let formattedMessage = offer.message
  if (offer.rawMessage && offer.rawMessage.text && offer.rawMessage.values) {
    formattedMessage = transformMessage(offer.rawMessage.text, offer.rawMessage.values)
  }

  return (
    <div className={css.offerWrapper}>
      <div className={classnames(css.discountOSR, css[getDiscountOSRClass(type)])}>
        <div className={css.discountOSR__container}>
          {offer.formattedValue}
          <small className={css.discountOSR__sub}>OFF</small>
        </div>
      </div>
      <div
        className={classnames(css.messageDiscountOSR, css[getMessageDiscountOSRClass(type)])}
        dangerouslySetInnerHTML={{ __html: formattedMessage }} // eslint-disable-line react/no-danger
      />
    </div>
  )
}

Offer.propTypes = propTypesOffer
Offer.defaultProps = defaultProps

export { Offer }
