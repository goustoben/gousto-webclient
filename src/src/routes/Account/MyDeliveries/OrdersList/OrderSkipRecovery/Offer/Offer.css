@value Haricot, White from 'styles/colors.css';
@value comfortaaFontStack from 'styles/typography.css';

.offerWrapper {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.discountOSR {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-shrink: 0;
  height: 78.44px;
  width: 78.77px;
  border: 1.44px solid #01A92B;
  background-color: Haricot;
  border-radius: 50%;
  box-sizing: border-box;
  line-height: 21px;
}

.discountOSR__container {
  display: flex;
  flex-direction: column;
  max-width: 60px;
  font-size: 22px;
  font-family: comfortaaFontStack;
  font-weight: bold;
  color: White;
}

.discountOSR__sub {
  font-size: 70%;
  line-height: 1.2;
}

.messageDiscountOSR {
  color: Haricot;
  font-family: Avenir;
  font-size: 16px;
  line-height: 21px;
  text-align: left;
  margin-left: 20px;
}