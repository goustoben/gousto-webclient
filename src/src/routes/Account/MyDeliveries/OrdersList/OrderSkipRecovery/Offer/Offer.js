import React, { PropTypes } from 'react'

import css from './Offer.css'

const propTypes = {
  offer: PropTypes.shape({
    formatted_value: PropTypes.string,
    raw_message: PropTypes.shape({
      text: PropTypes.string,
      values: PropTypes.shape({
        date: PropTypes.string,
        value: PropTypes.string,
      })
    })
  }).isRequired
}

const transformMessage = (text) => {
  // [firstPart][value][secondPart][date]
  const textSplitByValue = text.split('{:value:}')
  const firstPart = textSplitByValue[0]
  const textSplitByDate = textSplitByValue[1].split('{:date:}')
  const secondPart = textSplitByDate[0]

  return {
    firstPart,
    secondPart
  }
}

const Offer = ({ offer }) => {
	if (!offer) return null

  const offerValue = offer.raw_message.values.value
  const offerDate = offer.raw_message.values.date

  const formattedMessage = transformMessage(offer.raw_message.text, offerValue, offerDate)

  return (
    <div className={css.offerWrapper}>
      <div className={css.discountOSR}><div>{offer.formatted_value} OFF</div></div>
      <div className={css.messageDiscountOSR}>
        {formattedMessage.firstPart}
        <strong>{offerValue} OFF</strong>
        {formattedMessage.secondPart}
        <strong>{offerDate}</strong>
      </div>
    </div>
  )
}

Offer.propTypes = propTypes

export default Offer
