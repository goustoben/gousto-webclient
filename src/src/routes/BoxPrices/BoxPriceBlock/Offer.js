import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'

import css from './BoxPriceBlock.css'

export const Offer = ({
  icon,
  detailText,
  text,
}) => (
  <div className={css.offer}>
    <icon className={classNames(css.icon, css[icon])} />
    <p className={css.offerOption}>
      <bold className={css.offerOptionDetail}>{detailText}</bold>
      {text}
    </p>
  </div>
)

Offer.propTypes = {
  icon: PropTypes.string.isRequired,
  detailText: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
}
