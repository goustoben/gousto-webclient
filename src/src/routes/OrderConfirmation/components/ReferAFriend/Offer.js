import React from 'react'
import PropTypes from 'prop-types'
import { config } from './config.js'
import css from './ReferAFriend.css'

const propTypes = {
  isYourOffer: PropTypes.bool,
  offer: PropTypes.string,
  theirMonthOffer: PropTypes.string,
}

const Offer = ({ isYourOffer, offer, theirMonthOffer }) => {
  const offerConfig = config(theirMonthOffer)
  const title = isYourOffer ? offerConfig.youGetTitle : offerConfig.yourFriendGetsTitle
  const description = isYourOffer ? offerConfig.youGetDescription : offerConfig.yourFriendGetsDescription

  return (
    <section className={css.offerWrapper}>
      <div className={css.offerTitle}>{title}</div>
      <div className={css.offer}>{offer}</div>
      <div className={css.offerDescription}>{description}</div>
    </section>
  )
}

Offer.propTypes = propTypes

export { Offer }
