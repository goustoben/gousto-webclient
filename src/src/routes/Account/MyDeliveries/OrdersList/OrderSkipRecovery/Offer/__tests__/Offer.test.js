import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import Offer from '../Offer'

describe('Order Skip Recovery Model Offer', () => {
	let wrapper
	const offer = {
		message: 'You only have 10% on all your orders until the 19th of October',
		formattedValue: '10%',
		rawMessage: {
			text: 'You only have {:value:} on all your orders until the {:date:}',
			values: [
				{ key: 'date', value: '19th of October' },
				{ key: 'value', value: '£13' }
			]
		}
	}

	describe('Innitial Render', () => {
		beforeAll(() => {
			wrapper = mount(<Offer offer={offer} />)
		})

		test('should render snapshot', () => {
			const tree = renderer.create(
				<Offer offer={offer} />
			).toJSON()

			expect(tree).toMatchSnapshot()
		})

		test('should render discount amount', () => {
			const discountAmount = wrapper.find('.discountOSR')

			expect(discountAmount.text()).toBe('10% OFF')
		})

		test('should render discount message', () => {
			const discountMessage = wrapper.find('.messageDiscountOSR')

			expect(discountMessage.text()).toBe('You only have £13 on all your orders until the 19th of October')
		})
	})

	describe('Alternative Render', () => {
		test('should not render component if no offer prop is passed', () => {
			wrapper = mount(<Offer />)

			const wrapperDiv = wrapper.find('offerWrapper')

			expect(wrapperDiv.length).toBe(0)
		})
	})
})
