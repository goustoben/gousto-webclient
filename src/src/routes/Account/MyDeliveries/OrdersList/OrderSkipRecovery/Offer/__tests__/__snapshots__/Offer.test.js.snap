// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Skip Recovery Model Offer Innitial Render should render snapshot 1`] = `
<div
  className="offerWrapper"
>
  <div
    className="discountOSR"
  >
    <div
      className="discountOSR__container"
    >
      10%
      <small
        className="discountOSR__sub"
      >
        OFF
      </small>
    </div>
  </div>
  <div
    className="messageDiscountOSR"
    dangerouslySetInnerHTML={
      Object {
        "__html": "You only have <strong>Â£13</strong> on all your orders until the <strong>19th of October</strong>",
      }
    }
  />
</div>
`;
